src/main/scala/eventstore/EsProjectionsClient.scala

```diff
@@ -0,0 +1,280 @@
+  case class UnableToDeleteProjection(reason: String) extends ProjectionDeleteResult
+
+}
+
+/**
+ * A client allowing to create, get the status and delete an existing projection.
+ */
+class EsProjectionsClient(settings: Settings = Settings.Default, system: ActorSystem)(implicit materializer: Materializer) extends EventStoreProjectionsUrls {
+
+  import materializer.executionContext
+  import EsProjectionsClient._
+
+  private val connection: Flow[HttpRequest, Try[HttpResponse], NotUsed] = {
+    val uri = Uri(settings.http.url)
+    val httpExt = Http(system)
+
+    val connectionPool = if (settings.http.protocol == "http") {
+      httpExt.cachedHostConnectionPool[Unit](uri.authority.host.address(), uri.authority.port)
+    } else {
+      httpExt.cachedHostConnectionPoolHttps[Unit](uri.authority.host.address(), uri.authority.port)
+    }
+
+    Flow[HttpRequest]
+      .map(req => (req, ()))
+      .via(connectionPool)
+      .map { case (resp, _) => resp }
+  }
+
+  private val authorization: Option[Authorization] = settings.defaultCredentials.map(credentials => Authorization(BasicHttpCredentials(credentials.login, credentials.password)))
+
+  private val defaultHeaders = authorization.toList
+
+  /**
+   * Create the projection with the specified name and code
+   * @param name the name of the projection to create
+   * @param javascriptCode the javascript code for the projection
+   * @param mode the projection's mode (Either OneTime, Continuous or Transient)
+   * @param allowEmit
+   * @return
+   */
+  def createProjection(name: String, javascriptCode: String, mode: ProjectionMode = Continuous, allowEmit: Boolean = true): Future[ProjectionCreationResult] = {
+    val request = HttpRequest(
+      method = HttpMethods.POST,
+      uri = Uri(createProjectionUrl(name, mode, allowEmit)),
+      headers = defaultHeaders,
+      entity = HttpEntity(ContentTypes.`application/json`, javascriptCode))
+
+    singleRequestWithErrorHandling(request)
+      .map {
+        case response if response.status == StatusCodes.Created => ProjectionCreated
+        case response if response.status == StatusCodes.Conflict => ProjectionAlreadyExist
+        case response => throw new ProjectionException(s"Received unexpected reponse $response")
+      }
+      .runWith(Sink.head)
+  }
+
+  /**
+   * Fetch the status for the specified projection.
+   * @param name the name of the projection
+   * @return Absent if the projection doesn't exist. Otherwise either Running/Completed/Stopped/Faulted or Other.
+   */
+  def fetchProjectionStatus(name: String): Future[ProjectionStatus] = {
+    val request = HttpRequest(
+      method = HttpMethods.GET,
+      uri = Uri(projectionBaseUrl(name)),
+      headers = defaultHeaders)
+
+    singleRequestWithErrorHandling(request)
+      .mapAsync(1) {
+        case response if response.status == StatusCodes.NotFound => Future.successful(Absent)
+        case response => Unmarshal(response).to[String].map { rawJson =>
```

Comment:
maybe it does worth to create some case class for Projection with generic fields as status?